src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresDirectoryDAO.scala

```diff
@@ -469,6 +469,57 @@ class PostgresDirectoryDAO(protected val writeDbRef: DbReference, protected val
 
   override def updateUserEmail(userId: WorkbenchUserId, email: WorkbenchEmail, samRequestContext: SamRequestContext): IO[Unit] = IO.unit
 
+  override def updateUser(samUser: SamUser, userUpdate: UserUpdate, samRequestContext: SamRequestContext): IO[Option[SamUser]] =
+    serializableWriteTransaction("updateUser", samRequestContext) { implicit session =>
+      val u = UserTable.column
+
+      val setColumnsClause = userUpdate match {
+        case UserUpdate(None, None) => throw new WorkbenchException("Cannot update user with no values.")
+        case UserUpdate(Some(newAzureB2CId), None) =>
+          samsqls"(${u.azureB2cId}, ${u.updatedAt})"
+        case UserUpdate(None, Some(newGoogleSubjectId)) =>
+          samsqls"(${u.googleSubjectId}, ${u.updatedAt})"
+        case UserUpdate(Some(newGoogleSubjectId), Some(newAzureB2CId)) =>
+          samsqls"(${u.googleSubjectId}, ${u.azureB2cId}, ${u.updatedAt})"
+      }
+
+      var updatedUser =
+        samUser.copy(
+          googleSubjectId = userUpdate.newGoogleSubjectId.map(GoogleSubjectId).orElse(samUser.googleSubjectId),
+          azureB2CId = userUpdate.newAzureB2CId.map(AzureB2CId).orElse(samUser.azureB2CId),
+          updatedAt = Instant.now()
+        )
```

Comment:
I tried and couldnt get the update query to return the updated user, so this is the next best thing